test/functions/iterators/ldelaney/zipDynamic3d.chpl

```diff
@@ -0,0 +1,55 @@
+use DynamicIters;
```

Comment:
A concern I have with this test is that it wouldn't detect a bug where we accidentally yielded the same index twice. I don't think that would happen today, but if we can write this test in a way that would catch such an issue, that would be preferable.

I think another way to write this could be to declare an array of integers over the desired domain, and then in the loop body increment each index by one. After the loop, you can assert that each index was visited once.